src/Time/Formatting.hs

```diff
@@ -38,10 +50,27 @@ module Time.Formatting
 
 import Time.Rational (Rat)
 #if ( __GLASGOW_HASKELL__ >= 804 )
-import Time.Rational (withRuntimeDivRat)
+import Time.Rational (type (>=%), withRuntimeDivRat)
 #endif
 import Time.Units (AllTimes, KnownRatName, Time, floorUnit, toUnit)
 
+#if ( __GLASGOW_HASKELL__ >= 804 )
+-- Type-level 'if' for 'Bool's.
+type family If (c :: Bool) (t :: Bool) (e :: Bool) where
+  If 'True  t e = t
+  If 'False t e = e
+
+-- | Type family for verification of the descending order of the given
+-- list of time units.
+type family IsDescending (units :: [Rat]) :: Bool
+
+type instance IsDescending (unit1 ': unit2 ': units :: [Rat]) =
+    If (unit1 >=% unit2) (IsDescending (unit2 ': units :: [Rat]))
+                         'False
+type instance IsDescending (unit ': '[] :: [Rat]) = 'True
+type instance IsDescending ('[] :: [Rat]) = 'True
```

Comment:
I think you don't need to specify `:: [Rat]` in every case because you already specified type of `IsDescending` family argument.